Detections/OfficeActivity/MailItemsAccessed_TimeSeries.yaml

```diff
@@ -0,0 +1,64 @@
+id: b4ceb583-4c44-4555-8ecf-39f572e827ba
+name: Exchange workflow MailItemsAccessed operation anomaly
+description: |
+  'Identifies anomalous increases in Exchange mail items accessed operations. 
+  The query leverages KQL built-in anomaly detection algorithms to find large deviations from baseline patterns. 
+  Sudden increases in execution frequency of sensitive actions should be further investigated for malicious activity.
+  Manually change scorethreshold from 1.5 to 3 or higher to reduce the noise based on outliers flagged from the query criteria.
+  Read more about MailItemsAccessed: https://docs.microsoft.com/microsoft-365/compliance/advanced-audit?view=o365-worldwide#mailitemsaccessed'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: Office365
+    dataTypes:
+      - OfficeActivity
+queryFrequency: 1d
+queryPeriod: 14d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - Collection
+relevantTechniques:
+  - T1114.002
+query: |
+
+  let starttime = 14d;
+  let endtime = 1d;
+  let timeframe = 1h;
+  let scorethreshold = 1.5;
+  let percentthreshold = 50;
+  // Preparing the time series data aggregated hourly count of MailItemsAccessd Operation in the form of multi-value array to use with time series anomaly function.
+  let TimeSeriesData=
+  OfficeActivity 
+  | where TimeGenerated  between (startofday(ago(starttime))..startofday(ago(endtime)))
+  | where OfficeWorkload=~ "Exchange" and Operation =~ "MailItemsAccessed" and ResultStatus =~ "Succeeded"
+  | project TimeGenerated, Operation, MailboxOwnerUPN 
+  | make-series Total=count() on TimeGenerated from startofday(ago(starttime)) to startofday(ago(endtime)) step timeframe;
+  // Use the time series data prepared in previous step with time series aomaly function to generate baseline pattern and flag the outlier based on scorethreshold value.
+  let TimeSeriesAlerts = TimeSeriesData
+  | extend (anomalies, score, baseline) = series_decompose_anomalies(Total, scorethreshold, -1, 'linefit')
+  | mv-expand Total to typeof(double), TimeGenerated to typeof(datetime), anomalies to typeof(double), score to typeof(double), baseline to typeof(long)
+  | where anomalies > 0
+  | project TimeGenerated, Total, baseline, anomalies, score;
+  // Joining the flagged outlier from the previous step with the original dataset to present contextual information
+  // during the anomalyhour to analysts to conduct investigation or informed decisions.
+  TimeSeriesAlerts
+  | where TimeGenerated > ago(2d)
+  // Join against base logs since specified timeframe to retrive records associated with the hour of anomoly
+  | join ( 
+    OfficeActivity 
+  | where TimeGenerated between (startofday(ago(starttime))..startofday(ago(endtime)))
+  | where OfficeWorkload=~ "Exchange" and Operation =~ "MailItemsAccessed" and ResultStatus =~ "Succeeded"
```

Comment:
Could you add:
  | extend Client_IPAddress= split(Client_IPAddress,"]:")[0]
  | extend Client_IPAddress= iif(Client_IPAddress startswith "[", split(Client_IPAddress,"[")[1], ip)
this will standardize the Client_IPAddress to make it easier to pivot on values in that set.

Reply:
I`ll include it in another PR after troubleshooting build errors.